soda_machine.rb

```diff
@@ -7,12 +7,25 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
```

Comment:
Again, good logic, but excessive computation going on. Try re-implementing with the `Enumerable::find` method to make this a one-liner.